packages/cli/src/oclif/commands/register.js

```diff
@@ -63,37 +98,77 @@ class RegisterCommand extends ZapierBaseCommand {
     }
   }
 
+  /**
+   * Prompts user for values that have not been provided
+   * Flags can heavily impact the behavior of this function
+   * @returns {object}
+   */
   async _promptForAppMeta() {
     const appMeta = {};
 
+    const actionChoices = [
+      { name: 'Update current integration', value: 'update' },
+      { name: 'Register a new integration', value: 'register' },
+    ];
+    appMeta.action = actionChoices[1].value; // Default action is register
+    if (this._hasAppFile()) {
+      console.info(colors.yellow('.zapierapprc file detected.'));
+      if (this.flags.yes) {
+        console.info(
+          colors.yellow('-y/--yes flag passed, updating current integration.')
+        );
+        appMeta.action = actionChoices[0].value;
+      } else {
+        appMeta.action = await this.promptWithList(
+          'Would you like to update your current integration or register a new one?',
+          actionChoices
+        );
+      }
+    }
+
+    if (appMeta.action === 'update') {
+      this.startSpinner('Retrieving details for your integration');
+      this.app = await getWritableApp();
+      this.stopSpinner();
+    }
+
     appMeta.title = this.args.title?.trim();
     if (!appMeta.title) {
       appMeta.title = await this.prompt(
         'What is the title of your integration?',
-        { required: true }
+        {
+          required: true,
+          default: this.app?.title,
+        }
       );
     }
 
     appMeta.description = this.flags.desc?.trim();
     if (!appMeta.description) {
       appMeta.description = await this.prompt(
         `Please provide a sentence describing your app in ${MAX_DESCRIPTION_LENGTH} characters or less.`,
-        { required: true, charLimit: MAX_DESCRIPTION_LENGTH }
+        {
+          required: true,
+          charLimit: MAX_DESCRIPTION_LENGTH,
+          default: this.app,
+        }
       );
     }
 
     appMeta.homepage_url = this.flags.url;
-    if (!appMeta.homepage_url) {
+    if (!appMeta.homepage_url && !this.flags.yes) {
```

Comment:
I don't think we should skip `homepage_url` prompt when `-y` is enabled. Any reason why we may want to skip?

Reply:
@eliangcs I figured since it's optional, it should be skippable, but I'm okay not skipping it with `-y`.

Reply:
Updated in 915e630cc2cf6cf3b7958ddac772012dda29b6e0